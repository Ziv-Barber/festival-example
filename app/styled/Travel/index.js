// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars
import styled from 'styled-components'

// Other components that we need:
import { ItemsList } from '../../components/ItemsList'
import TravelBox from '../TravelBox'

// Content that we need:
import travelList from '../../content/travel'

// Create the list of items for the Lineup component:
const travelItems = travelList.map((value) => ({
  key: value.key,
  component: TravelBox,
  compProps: {
    name: value.name,
    image: value.image,
    options: value.options
  }
  // gridProps
}))

/**
 * Tickets react component.
 */
export default styled(ItemsList).attrs({
  items: travelItems,
  caption: 'Travel',
  captionProps: '',
  captionVariant: '',
  containerId: '',
  gridRootDefProps: {
    spacing: 8,
    justify: 'space-around',
    alignItems: 'center'
  },
  gridDefProps: {
    xs: 12,
    sm: 4,
    md: 3,
    lg: 2,
    xl: 1
  }
})`
  margin: 0 !important;
  padding: ${(props) => props.theme.spacing.unit}px 0 !important;
  width: 100% !important;
  background: linear-gradient(to bottom, #4d94ff, #0052cc);
  color: white;

  & .listHeader {
    padding: ${(props) => props.theme.spacing.unit}px !important;
    width: 100% !important;
    text-shadow: 2px 2px 5px #fff !important;
    text-align: center;
    color: #003366;
  }

  & .listGrid {
    margin: 0 !important;
    width: 100% !important;
    padding: ${(props) => props.theme.spacing.unit}px !important;
  }

  & .listItem {
    margin: ${(props) => props.theme.spacing.unit}px !important;
    z-index: 1050;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  & .card {
    min-width: 254px;
    background-image: none;
    background-color: transparent;
  }

  & .cardImage {
    width: 250px;
    border: 2px solid #fff;
    padding: 0 !important;
  }

  & .cardContent {
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 8px !important;
  }

  & .cardActions {
    position: relative;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 0 !important;
  }

  & .cardInfo {
    position: absolute;
    top: 46px;
    left: 100px;
  }

  & h2 {
    color: #fff !important;
  }

  & .popup .paper {
    width: 80vw;
    min-height: 300px;
  }
`
