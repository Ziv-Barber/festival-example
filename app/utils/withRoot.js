// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars
import { MuiThemeProvider } from '@material-ui/core/styles'
import { ThemeProvider } from 'styled-components'
import CssBaseline from '@material-ui/core/CssBaseline'
import theme from '../theme'

/**
 * withRoot HOC to wrap a react component with the app's theme provider.
 */
const withRoot = (Component: React.ComponentType<any>) => {
  const withRootBase = (props: any) => (
    <MuiThemeProvider theme={theme}>
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <CssBaseline />
          <Component {...props} />
        </React.Fragment>
      </ThemeProvider>
    </MuiThemeProvider>
  )

  // $FlowFixMe
  withRootBase.displayName = `withRoot(${Component.displayName ||
    Component.name})`

  return withRootBase
}

export default withRoot
