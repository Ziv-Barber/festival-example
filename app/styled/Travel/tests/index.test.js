// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import { render, cleanup } from 'react-testing-library'

import Travel from '../index'
import withRoot from '../../../utils/withRoot'

// For flow:
let { describe, it, afterEach, expect } = global

/** @test {Travel} */
describe('<Travel />', () => {
  afterEach(cleanup)

  /** @test {Travel} */
  it('should render', () => {
    const App = withRoot(Travel)
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })
})
