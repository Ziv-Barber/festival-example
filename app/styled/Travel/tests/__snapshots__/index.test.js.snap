// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Travel /> should render 1`] = `
<div>
  <div
    class="MuiPaper-root-1 MuiPaper-elevation0-3 Travel-nizlnn-0 JnxXn"
  >
    <h3
      class="MuiTypography-root-28 MuiTypography-h3-42 listHeader"
    >
      Travel
    </h3>
    <div
      class="MuiGrid-container-64 MuiGrid-spacing-xs-8-86 MuiGrid-align-items-xs-center-72 MuiGrid-justify-xs-space-around-84 listGrid"
    >
      <div
        class="MuiGrid-item-65 MuiGrid-grid-xs-12-104 MuiGrid-grid-sm-4-110 MuiGrid-grid-md-3-123 MuiGrid-grid-lg-2-136 MuiGrid-grid-xl-1-149 listItem"
      >
        <div
          class="MuiPaper-root-1 MuiPaper-elevation0-3 MuiCard-root-161 card"
        >
          <div
            class="MuiCardContent-root-162 cardContent"
          >
            <h2
              class="MuiTypography-root-28 MuiTypography-h6-45"
            >
              By 
              bus
            </h2>
          </div>
          <div
            class="MuiCardActions-root-163 cardActions"
          >
            <div
              class="MuiCardActions-action-165"
            >
              <img
                class="cardImage"
                src="IMAGE_MOCK"
              />
              <div
                class="cardInfo"
              >
                <button
                  class="MuiButtonBase-root-176 MuiFab-root-166 MuiFab-primary-168"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiFab-label-167"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root-179"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root-204"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-item-65 MuiGrid-grid-xs-12-104 MuiGrid-grid-sm-4-110 MuiGrid-grid-md-3-123 MuiGrid-grid-lg-2-136 MuiGrid-grid-xl-1-149 listItem"
      >
        <div
          class="MuiPaper-root-1 MuiPaper-elevation0-3 MuiCard-root-161 card"
        >
          <div
            class="MuiCardContent-root-162 cardContent"
          >
            <h2
              class="MuiTypography-root-28 MuiTypography-h6-45"
            >
              By 
              train
            </h2>
          </div>
          <div
            class="MuiCardActions-root-163 cardActions"
          >
            <div
              class="MuiCardActions-action-165"
            >
              <img
                class="cardImage"
                src="IMAGE_MOCK"
              />
              <div
                class="cardInfo"
              >
                <button
                  class="MuiButtonBase-root-176 MuiFab-root-166 MuiFab-primary-168"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiFab-label-167"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root-179"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root-204"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-item-65 MuiGrid-grid-xs-12-104 MuiGrid-grid-sm-4-110 MuiGrid-grid-md-3-123 MuiGrid-grid-lg-2-136 MuiGrid-grid-xl-1-149 listItem"
      >
        <div
          class="MuiPaper-root-1 MuiPaper-elevation0-3 MuiCard-root-161 card"
        >
          <div
            class="MuiCardContent-root-162 cardContent"
          >
            <h2
              class="MuiTypography-root-28 MuiTypography-h6-45"
            >
              By 
              plane
            </h2>
          </div>
          <div
            class="MuiCardActions-root-163 cardActions"
          >
            <div
              class="MuiCardActions-action-165"
            >
              <img
                class="cardImage"
                src="IMAGE_MOCK"
              />
              <div
                class="cardInfo"
              >
                <button
                  class="MuiButtonBase-root-176 MuiFab-root-166 MuiFab-primary-168"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiFab-label-167"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root-179"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root-204"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
